src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -472,31 +464,37 @@ void CARMACloudPlugin::OnStateChange(IvpPluginState state) {
 	}
 }
 
-
-int CARMACloudPlugin::CloudSend(string msg,string url, string base, string method)
+void CARMACloudPlugin::CloudSendAsync(const string& local_msg,const string& local_url, const string& local_base, const string& local_method)
 {
-	CURL *req;
-  	CURLcode res;
-	string urlfull = url+base;	  
-
+	std::thread t([this, &local_msg, &local_url, &local_base, &local_method](){	
+		CloudSend(local_msg, local_url, local_base, local_method);	
+	});
+	t.detach();
+}
 
-  	req = curl_easy_init();
- 	 if(req) {
-  	  	curl_easy_setopt(req, CURLOPT_URL, urlfull.c_str());
+int CARMACloudPlugin::CloudSend(const string &local_msg, const string& local_url, const string& local_base, const string& local_method)
+{ 	
+	CURL *req;
+	CURLcode res;
+	string urlfull = local_url+local_base;	
+	req = curl_easy_init();
+	if(req) {
+		curl_easy_setopt(req, CURLOPT_URL, urlfull.c_str());
 
-		if(strcmp(method.c_str(),"POST")==0)
+		if(strcmp(local_method.c_str(),"POST")==0)
 		{
-    		curl_easy_setopt(req, CURLOPT_POSTFIELDS, msg.c_str());
-			curl_easy_setopt(req, CURLOPT_TIMEOUT, 2L); // Sets a 2 second timeout 
+			curl_easy_setopt(req, CURLOPT_POSTFIELDS, local_msg.c_str());
+			curl_easy_setopt(req, CURLOPT_TIMEOUT_MS, 1000L); // Request operation complete within max millisecond timeout 

```

Comment:
The comment says request operation to be complete "within max millisecond" but the value is 1000L which is a second.  I think the comment is misleading,  Please clarify.